src/v/cloud_storage/remote_segment.cc

```diff
@@ -273,6 +276,32 @@ remote_segment::data_stream(size_t pos, ss::io_priority_class io_priority) {
     co_return storage::segment_reader_handle(std::move(data_stream));
 }
 
+ss::future<stream_with_leased_client> remote_segment::build_stream() {
+    retry_chain_node local_rtc(
+      cache_hydration_timeout, cache_hydration_backoff, &_rtc);
+
+    stream_with_leased_client slc;
+    auto res = co_await _api.download_segment(
```

Comment:
I think that this may kill the network because every fetch will start a new GET request and buffer a lot of memory. In case if readers are reused its OK but if they're not we will read much more data than we're consuming. For instance, if we have a wide fetch request that fetches from 1000 partitions and each stream fetches 1MiB we will need 1GiB of shard memory to accommodate read buffers. If we don't have enough it will start throwing old readers away so the next fetch will start new readers and will also read 1GiB and use only first 500MiB. This will double the net bandwidth used to serve the request and a lot of data will be in-flight at any moment.

It should be possible to improve this by passing byte range to the GET request explicitly. The value can probably be computed based on index. In the `log_reader_config` we have a `max_bytes` field which can be used for this. But the problem is that our indexes are not precise enough for this trick.

Reply:
> In case if readers are reused its OK but if they're not we will read much more data than we're consuming

I ended up keeping the reader's parser and stream open while it is cached. It was much more expensive and slower to reset the reader after the client goes away. 

So the reader keeps the connection open. For this exploration it worked fine in OMB runs but I think for such readers connected to an HTTP endpoint rather than disk we need to timeout and reset them because after a certain time the cloud storage HTTP server may close the connection and the reader is not reusable after that.